plugins/logging/file

```diff
@@ -279,13 +279,14 @@ sub hook_logging {
     if (   !$self->{_f}
         || !$self->{_nosplit}
         || !$transaction
+        || !UNIVERSAL::can($transaction, 'notes')
         || !$transaction->notes('file-logged-this-session'))
     {
         unless (defined $self->maybe_reopen($transaction)) {
             return DECLINED;
         }
-        if (UNIVERSAL::can($transaction,'isa')) {
-            $transaction->notes('file-logged-this-session', 1) if $transaction;
+        if (UNIVERSAL::can($transaction,'isa') && $transaction && UNIVERSAL::can($transaction, 'notes')) {
```

Comment:
The ordering here is incorrect. You have to check it $transaction before you check if it's a reference.
```diff
- if (UNIVERSAL::can($transaction,'isa') && $transaction && UNIVERSAL::can($transaction, 'notes')) {
+ if ($transaction && UNIVERSAL::can($transaction,'isa') && UNIVERSAL::can($transaction, 'notes')) {
```